src/server/main_service.cc

```diff
@@ -1066,12 +1066,14 @@ bool Service::InvokeCmd(const CommandId* cid, CmdArgList tail_args, ConnectionCo
     DispatchMonitor(cntx, cid, tail_args);
   }
 
+  cntx->reply_builder()->ExpectReply();
   try {
     cid->Invoke(tail_args, cntx);
   } catch (std::exception& e) {
     LOG(ERROR) << "Internal error, system probably unstable " << e.what();
     return false;
   }
+  DCHECK(cntx->reply_builder()->HasReplied());
 
```

Comment:
would be my solution suggestion, not sure if its simpler 🙂 

```
#ifdef NDEBUG
CapturingReplyBuilder crb{ReplyMode::ALL};
auto* old = cntx->Inject(&crb);
#endif

*invoke*

#ifdef NDEBUG
auto reply = crb.Take();
DCHECK(reply.index() > 0);
CRB::Apply(reply, (*cntx).operator->());
#endif

Reply:
I have some of the mechanics as a convenient RAII in 1924, you could steal it from there if you choose this approach

Reply:
It would not work with memcached, because the `Apply` function accepts a `RedisReplyBuilder`. Moreover, for some reason it made some of the tests stack. 

However, I really like this idea, IMO, it's cleaner. What I will try on a seprate PR (because I don't want to block us) is:

1. Overload `Apply` to work with MCBuilder (or just make it accept a SinkReplyBuilder). 
2. Figure out what cause the hangs in the tests.

Reply:
I'd suggest just doing this for non memcached commands because CRB has no support for storing mc-like replies

Reply:
Other way round its likely not worth the effort

Reply:
> I'd suggest just doing this for non memcached commands because CRB has no support for storing mc-like replies 

So just `skip` for the `memcached` commands? I can do this by `dyn_cast` and checking if the `reply builder` is `MCBuilder`. I will try this out

Reply:
But second thoughts we would also miss the `dcheck` for `memcached` commands which I am not really happy about... @romange any opinion on this?

Reply:
@dranikpg I applied your comments, I don't want to delay this any longer. Took a little bit more time, because I had to add a few exceptions but it passes locally

Reply:
I do not care about missing memcache - it has a small surface and it does not change a lot.

Reply:
i saw now previous commits. I prefer the boolean option